lib/publisher.js

```diff
@@ -45,31 +47,22 @@ class Publisher {
 
   /**
    * Publish a message
-   * @param {Buffer} message - A Node.js buffer object, representing the message to be published.
-   * @return {undefined}
+   * @param {Object} message - An object of the topic message.
+   * @return {void}
    */
   publish(message) {
-    let msg;
-    if (message instanceof Buffer) {
-      msg = message;
-    } else if (typeof message.getBuffer === 'function') {
-      msg = message.getBuffer();
-    }  // TODO(kenny): check messageType and convert raw values to a ros message
-
-    if (msg) {
-      rclnodejs.publishMessage(this._handle, msg);
+    let rawRosMessage = this._typeClass.getRefBuffer(message);
+    if (rawRosMessage) {
```

Comment:
It does improve the flexibility by supporting something like:

`publisher.publish('plain string')` or `publisher.publish({x: 1, y: 2, z: 3})`

Good suggestion! I will mark this as a TODO, so we can implement it later.